dd-trace-propagation/src/tracecontext.rs

```diff
@@ -84,85 +84,117 @@ fn inject_traceparent(context: &SpanContext, carrier: &mut dyn Injector) {
 }
 
 fn inject_tracestate(context: &SpanContext, carrier: &mut dyn Injector) {
-    let mut tracestate_parts = vec![];
+    // Use a single String buffer to build the entire tracestate, avoiding intermediate allocations
+    let mut dd_parts = String::new();
 
+    // Build sampling priority part
     let priority = context.sampling.priority.unwrap_or(priority::USER_KEEP);
+    dd_parts.push_str(TRACESTATE_SAMPLING_PRIORITY_KEY);
+    dd_parts.push(':');
+    dd_parts.push_str(&priority.to_string());
 
-    tracestate_parts.push(format!("{TRACESTATE_SAMPLING_PRIORITY_KEY}:{priority}"));
-
-    if let Some(origin) = context.origin.as_ref().map(|origin| {
-        encode_tag_value(
+    // Build origin part if present
+    if let Some(origin) = context.origin.as_ref() {
+        let origin_encoded = encode_tag_value(
```

Comment:
`encode_tag_value` always copies the string, even if not modifications are done.
You could replace it with something like 
```rust
fn encode_tag_value(tag: &str) -> Cow<'_, str> {
  if tag.as_bytes().contains(b'=') {
    Cow::Owned(tag.replace('=', "~"))
  } else {
    Cow::Borrowed(tag)
  }
}
```